composer/datasets/streaming/dataset.py

```diff
@@ -92,6 +93,10 @@ class StreamingDataset(IterableDataset):
         timeout (float): How long to wait for shard to download before raising an exception. Default: 60 sec.
         batch_size (Optional[int]): Hint the batch_size that will be used on each device's DataLoader. Default:
             ``None``.
+        shuffle_size (Optional[str]): Either a proportion of the dataset that is required to be loaded at once for
+            optimal streaming speed, or an absolute number of shards. The string takes the format: ``0.01prop``, meaning
+            1% of the dataset at a time is needed, or ``1000shards``, meaning 1000 shards at a time are needed. Lower values
+            may degrade shuffle randomness, while higher values may increase cold-start lag.
```

Comment:
I'm still chewing on this, but it seems that if proportion is always a float between [0.0,1.0] and shard count is always an integer in [1, inf), then you don't need to have the suffix like `prop` and `shards`, you can just case based on the dtype (float or int)

Reply:
Ravi suggested that this could potentially lead to unexpected behavior if the user accidentally typed `1.` instead of `1` and totally changed shuffling quasi randomness, whereas manually specifying the units requires you to commit to a shard/proportion decision.